tests/unit/env_test.yaml

```diff
@@ -0,0 +1,252 @@
+      - isNull: *is_null_env
+        documentIndex: 2
+      - isNull: *is_null_env
+        documentIndex: 3
+      - isNull: *is_null_env
+        documentIndex: 4
+      - isNull: *is_null_env
+        documentIndex: 5
+      - isNull: *is_null_env
+        documentIndex: 6
+      - isNull: *is_null_env
+        documentIndex: 7
+      - isNull: *is_null_env
+        documentIndex: 8
+      - isNull: *is_null_env
+        documentIndex: 9
+      - isNull: *is_null_env
+        documentIndex: 10
+      - isNull: *is_null_env
+        documentIndex: 11
+      - isNull: *is_null_env
+        documentIndex: 13
+
+  - it: Jobs default to no env
+    template: jobs.yaml
+    set:
+      st2:
+        rbac: { enabled: true } # enable rbac job
+    asserts:
+      - hasDocuments:
+          count: 4
+      - isNull: *is_null_env
+
+  - it: Deployments accept custom env
+    template: deployments.yaml
+    set:
+      st2auth:
+        env: &env
+          FOO: bar
+      st2api:
+        env: *env
+      st2stream:
+        env: *env
+      st2web:
+        env: *env
+      st2rulesengine:
+        env: *env
+      st2timersengine:
+        env: *env
+      st2workflowengine:
+        env: *env
+      st2scheduler:
+        env: *env
+      st2notifier:
+        env: *env
+      st2sensorcontainer:
+        env: *env
+      st2: { packs: { sensors: [] } } # ensure only 1 sensor
+      st2actionrunner:
+        env: *env
+      st2garbagecollector:
+        env: *env
+      st2client:
+        env: *env
+      st2chatops:
+        enabled: false
+    asserts:
+      - hasDocuments:
+          count: 13 # st2chatops env handled as secrets
+
+      - contains: &contains_env
+          path: spec.template.spec.containers[0].env
+          content:
+            name: FOO
+            value: bar
+
+  - it: Jobs accept custom env
+    template: jobs.yaml
+    set:
+      st2:
+        rbac: { enabled: true } # enable rbac job
+      jobs:
+        env: *env
+    asserts:
+      - hasDocuments:
+          count: 4
+
+      - contains: *contains_env
+
+  - it: Deployments envFrom defaults include st2-urls configMap and st2chatops secretRef
+    template: deployments.yaml
+    set:
+      st2: { packs: { sensors: [] } } # ensure only 1 sensor
+      st2chatops:
+        enabled: true
+    release:
+      name: st2ha
+    asserts:
+      - hasDocuments:
+          count: 14
+      - contains: &contains_st2_urls
+          path: spec.template.spec.containers[0].envFrom
+          content:
+            configMapRef:
+              name: st2ha-st2-urls
+        documentIndex: 0
+      - contains: *contains_st2_urls
+        documentIndex: 1
+      - contains: *contains_st2_urls
+        documentIndex: 2
+      - contains:
+          path: spec.template.spec.containers[0].envFrom
+          content:
+            configMapRef:
+              name: st2ha-st2-urls
+              optional: true # TODO: consider dropping this unique diff from other deployments
```

Comment:
We might want to drop this `optional: true` line from the st2web deployment as it makes the tests more complicated.
@armab It's been a long time, but do you remember why this was marked optional for st2web but not for anything else?

Reply:
https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/
> #### Optional References
> A ConfigMap reference may be marked "optional". If the ConfigMap is non-existent, the mounted volume will be empty. If the ConfigMap exists, but the referenced key is non-existent the path will be absent beneath the mount point.
>
> You must create a ConfigMap before referencing it in a Pod specification (unless you mark the ConfigMap as "optional").

No, don't remember why would we need it.
Maybe something was missed during the enterprise -> oss transition.

I'm good dropping it :+1:

Reply:
dropped.